src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientSkinSystem.java

```diff
@@ -15,38 +15,37 @@
  */
 package org.terasology.ligthandshadow.componentsystem.controllers;
 
-import org.terasology.entitySystem.entity.EntityBuilder;
-import org.terasology.entitySystem.entity.EntityManager;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.ReceiveEvent;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.ligthandshadow.componentsystem.LASUtils;
 import org.terasology.ligthandshadow.componentsystem.events.AddPlayerSkinToPlayerEvent;
-import org.terasology.logic.location.LocationComponent;
-import org.terasology.registry.In;
+import org.terasology.logic.characters.VisualCharacterComponent;
+import org.terasology.rendering.logic.SkeletalMeshComponent;
+import org.terasology.utilities.Assets;
 
 @RegisterSystem(RegisterMode.CLIENT)
 public class ClientSkinSystem extends BaseComponentSystem {
-    @In
-    private EntityManager entityManager;
-
-    private EntityBuilder builder;
 
+    /**
+     * Changes a player's skin on receiving the corresponding event
+     */
     @ReceiveEvent
     public void onAddPlayerSkinToPlayer(AddPlayerSkinToPlayerEvent event, EntityRef entity) {
```

Comment:
In this case is `entity` distinct from `event.player` ? Just curious. If it is you could add `VisualCharacterComponent` as a method parameter to both automatically filter by having it present (skipping one `if` statement) and assigning in to a variable (skipping a line to then declare a var normally)

Reply:
Yes, the receiving entities are client entities, hence it is necessary to have a player entity in the event for changing the skin of that particular player.